datastore/client/datastore_client.go

```diff
@@ -90,6 +93,52 @@ func (client DatastoreClient) Upsert(key string, value string) error {
 	return nil
 }
 
-func (client DatastoreClient) getUrlForKey(key string) string {
+func (client *DatastoreClient) getUrlForKey(key string) string {
 	return fmt.Sprintf("http://%v:%v/%v/%v", client.ipAddr, client.port, keyEndpoint, key)
 }
+
+/*
+Wait for healthy response
+*/
+func (client *DatastoreClient) WaitForHealthy(retries uint32, retriesDelayMilliseconds int) error {
```

Comment:
Same guidance around making the API self-guiding: it wouldn't make sense for `retriesDelayMilliseconds` to be negative, so we should prevent users from even doing it by using type `uint32`